source/grid/manifold.cc

```diff
@@ -630,6 +630,35 @@ FlatManifold<dim, spacedim>::get_new_point(
 
 
 
+namespace
+{
+  template <int n_eval_points, int spacedim>
+  std::array<Point<spacedim>, n_eval_points>
+  interpolate_points(const unsigned int         n_points,
+                     const Point<spacedim> *    surrounding_points,
+                     const double *             weights,
```

Comment:
The compiler might still think that these two pointers could alias or not? Or is this irrelevant, because they are all only read from?

Reply:
`new_point` used to be function local and is the only variable to which one adds. Why does the compiler have aliasing issues?

Reply:
This is one of the more funny situations. What happens is that the local variable `new_point` causes the aliasing. We have `boost::small_vector`, then conditionally take the address of that pointer. Since the small vector can be on the stack or on the heap, the compiler (erroneously) loses track of the dimension of the loop that accesses the array, which is the same condition as when the small vector lives on the stack. In the later optimization passes, it sees that we access variables on the stack through a plain pointer, it sees the `new_point` variable in the same memory region and thus decides to be conservative. I believe this is compiler-specific (looking at clang-15). Anyway, it disappears once we go out to a separate function, because the stack variables in that function cannot alias with a pointer in another.

Reply:
Could you put a description of this kind into the documentation of this function? The existence of the function is not obvious otherwise.